src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Routing;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Resource\ResourceInterface;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Helps add and import routes into a RouteCollection..
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class RouteCollectionBuilder
+{
+    private $loader;
+
+    /**
+     * A mixture of different objects that hold routes.
+     *
+     * @var Route[]|RouteCollectionBuilder[]|RouteCollection[]
+     */
+    private $routes = array();
+
+    private $defaults = array();
+
+    private $prefix;
+
+    private $host;
+
+    private $condition;
+
+    private $requirements = array();
+
+    private $options = array();
+
+    private $schemes;
+
+    private $methods;
+
+    private $resources = array();
+
+    private $controllerClass;
+
+    /**
+     * @param Loader $loader
+     */
+    public function __construct(Loader $loader)
```

Comment:
Like in #15742, this could be `LoaderInterface` instead... which I believe would be ok, since I think we could still "import", just using the longer syntax below (which the user wouldn't need to be bothered with).